runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -203,6 +215,48 @@ private boolean checkCoreArchitecture() {
         return true;
     }
 
+    private boolean verifyPackageSignature(String packageName, String fingerprint) {
+        Signature[] signatures = null;
+        try {
+            PackageManager packageManager = XWalkApplication.getApplication().getPackageManager();
+            PackageInfo packageInfo = packageManager.getPackageInfo(
+                    packageName, PackageManager.GET_SIGNATURES);
+            signatures = packageInfo.signatures;
+        } catch (NameNotFoundException e) {
+            Log.d(TAG, packageName + " not found");
+            return false;
+        }
+
+        if (signatures == null) {
+            Log.d(TAG, "No signature in package");
+            return false;
+        }
+
+        for (int i = 0; i < signatures.length; ++i) {
+            Log.d(TAG, "Checking signature " + i);
+            byte[] signature = signatures[i].toByteArray();
+
+            try {
+                MessageDigest md = MessageDigest.getInstance("SHA1");
```

Comment:
And to make it more confusing, it does seem that the rest of things *are* sha256, it's just the hashing algorithm that defaults to sha1